packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec

```
@@ -1,127 +1,75 @@
 
+# start specfile generated dependencies
 Requires: salt-master
-%if %{with python2}
-Requires: python
-BuildRequires: python2-rpm-macros
-%endif
-%if %{with python3}
 Requires: python3
 BuildRequires: python3-rpm-macros
-%endif
-
-# start specfile generated dependencies
 Requires: foreman-proxy >= %{foreman_proxy_min_version}
-Requires: %{?scl_prefix_ruby}ruby(release)
-Requires: %{?scl_prefix_ruby}ruby
-Requires: %{?scl_prefix_ruby}ruby(rubygems)
-Requires: %{?scl_prefix}rubygem(smart_proxy_dynflow) >= 0.5.0
-BuildRequires: %{?scl_prefix_ruby}ruby(release)
-BuildRequires: %{?scl_prefix_ruby}ruby
-BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+Requires: ruby >= 2.7
+Requires: ruby < 4
+BuildRequires: ruby >= 2.7
+BuildRequires: ruby < 4
+BuildRequires: rubygems-devel
 BuildArch: noarch
-Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 Provides: foreman-proxy-plugin-%{plugin_name} = %{version}
 # end specfile generated dependencies
 
-%{?scl:Obsoletes: rubygem-%{gem_name} < %{version}-%{release}}
-
 %description
-SaltStack Plug-In for Foreman's Smart Proxy.
+This plug-in adds support for Salt to Foreman's Smart Proxy.
 
 
 %package doc
 Summary: Documentation for %{name}
-Group: Documentation
 Requires: %{name} = %{version}-%{release}
 BuildArch: noarch
 
-%{?scl:Obsoletes: rubygem-%{gem_name}-doc < %{version}-%{release}}
-
 %description doc
 Documentation for %{name}.
 
 %prep
-%{?scl:scl enable %{scl} - << \EOF}
-gem unpack %{SOURCE0}
-%{?scl:EOF}
-
-%setup -q -D -T -n  %{gem_name}-%{version}
-
-%if %{with python2} && 0%{?rhel} >= 8
-sed -i -e '1s|^#!.*$|#!%{__python2}|' sbin/upload-salt-reports
-%endif
-
-%if %{with python3} && 0%{?rhel} >= 8
+%setup -q -n %{gem_name}-%{version}
 sed -i -e '1s|^#!.*$|#!%{__python3}|' sbin/upload-salt-reports
-%endif
-
-%{?scl:scl enable %{scl} - << \EOF}
-gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
-%{?scl:EOF}
 
 %build
 # Create the gem as gem install only works on a gem file
-%{?scl:scl enable %{scl} - << \EOF}
-gem build %{gem_name}.gemspec
-%{?scl:EOF}
+gem build ../%{gem_name}-%{version}.gemspec
 
 # %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
 # by default, so that we can move it into the buildroot in %%install
-%{?scl:scl enable %{scl} - << \EOF}
 %gem_install
-%{?scl:EOF}
 
 %install
 mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* %{buildroot}%{gem_dir}/
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
 
+mkdir -p %{buildroot}%{_bindir}
+cp -a .%{_bindir}/* \
+        %{buildroot}%{_bindir}/
 find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
-sed -ri 'sX.*/usr/bin/ruby|/usr/bin/env ruby.*$X\#\!/usr/bin/%{?scl:%{scl_prefix}}rubyX' .%{_bindir}/foreman-node
-sed -ri 'sX.*/usr/bin/ruby|/usr/bin/env ruby.*$X\#\!/usr/bin/%{?scl:%{scl_prefix}}rubyX' .%{_bindir}/salt_python_wrapper
+
+sed -ri 'sX.*/usr/bin/ruby|/usr/bin/env ruby.*$X\#\!/usr/bin/rubyX' .%{_bindir}/foreman-node
```

It's about this line: https://github.com/theforeman/smart_proxy_salt/blob/924fcfb7a3739cd4b62cd01c55a4f618d98e8db1/bin/foreman-node#L1

I think the current replacement doesn't make a lot of sense, because it'd likely attempt to replace `/usr/bin/ruby` with `/usr/bin/ruby`. You can probably already simplify it to:
```suggestion
sed -ri 'sX.*/usr/bin/env ruby.*$X\#\!/usr/bin/rubyX' .%{_bindir}/foreman-node
```

Or even simplify it further and tell it to remove `env ` from the first line:
```suggestion
sed -i '1 s/env //' .%{_bindir}/foreman-node
```

Though I don't know if RPM macros already do that anyway.